### PR TITLE
COMMERCE-12062, COMMERCE-12091

### PR DESCRIPTION
**Pull request by brianikim**
https://liferay.atlassian.net/browse/COMMERCE-12062
https://liferay.atlassian.net/browse/COMMERCE-12091

COMMERCE-12062:
1. On Discount Publish, check for Cart Total rules and check if Cart Total Minimum Amount is empty and throw exception and validate. 

COMMERCE-12091:
1. Attempt to get BigDecimal, which will fail and we handle it through MVCActionCommand instead and throw exception and validate.

@luca-pellizzon , @alessiorendina

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/service/impl/CommerceDiscountLocalServiceImpl.java [hunk 1]
```diff
@@ -1977,6 +1981,26 @@ private void _validate(
...
+				_commerceDiscountRuleLocalService.getCommerceDiscountRules(
+					commerceDiscountId, QueryUtil.ALL_POS, QueryUtil.ALL_POS,
+					null);
+
+			for (CommerceDiscountRule commerceDiscountRule :
+					commerceDiscountRules) {
+
+				String type = commerceDiscountRule.getType();
+
+				if (type.equals(
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/service/impl/CommerceDiscountLocalServiceImpl.java:1996]: If you do
`CommerceDiscountRuleConstants.TYPE_CART_TOTAL.equals(commerceDiscountRule.getType())`
you don't need the `type` variable

**Review by luca-pellizzon — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
I approved this PR even though I don't really like the solution.
We added in the service layer of discounts a validation based on discount rules. What if someone wants to implement a different rule as a custom implementation? They can't validate it in the discounts service

**Comment by alessiorendina** [comment]
Agree, that's why I suggested to have the validation on the rule and call directly the implementation based on the type in the service

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
@luca-pellizzon , @alessiorendina , 

new PR should satisfy the concerns you guys have raised. Implemented `validate` method directly into `*RuleTypeImpl` so we can support customization.

**Comment by brianikim** [comment]
@luca-pellizzon , let me know if this could proceed with QA still 

cc/ @pietrobocale

Sincerely,
Brian I. Kim

**Review by luca-pellizzon — APPROVED** [review]

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:close

**Pull request closed without merging**